File: src/cs/Schematron.Insight/ExportFormats.cs
using System;
using System.ComponentModel.DataAnnotations;


namespace Schematron.Insight
{
    [Flags]
    public enum ExportFormats : int
    {
        /// <summary>
        /// Formatted log text
        /// Serialize can available, but Deserialize cannot.
        /// </summary>
        [Display(Name = "Log", Description = "Serialize can available, but Deserialize cannot.")]
        Log = 0,
        /// <summary>
        /// Tab split text
        /// Serialize can available, but Deserialize cannot.
        /// </summary>
        [Display(Name = "Tab", Description = "Serialize can available, but Deserialize cannot.")]
        Tab = 1 << 0,
        /// <summary>
        /// markup text
        /// Serialize/Deserialize can available
        /// </summary>
        [Display(Name = "Xml", Description = "Serialize/Deserialize can available")]
        Xml = 1 << 1,
        /// <summary>
        /// json
        /// Serialize/Deserialize can available
        /// </summary>
        [Display(Name = "Json", Description = "Serialize/Deserialize can available")]
        Json = 1 << 2,
        /// <summary>
        /// Html
        /// Serialize can available, but Deserialize cannot.
        /// </summary>
        [Display(Name = "Html", Description = "Serialize can available, but Deserialize cannot.")]
        Html = 1 << 3
    }
}

File: src/java/Schematron.Insight/src/main/java/jp/gr/java_conf/schkit/enums/ExportFormats.java
package jp.gr.java_conf.schkit.enums;

public enum ExportFormats {
	None(0, ""),
	Log(1 << 0, "Log"),
	Tab(1 << 1, "Tab"),
	Xml(1 << 2, "Xml"),
	Json(1 << 3, "Json"),
	Html(1 << 4, "Html");

	private final int id;
	private final String name;
	private ExportFormats(int id, String name)
	{
		this.id = id;
		this.name= name;
	}
	public int getId()
	{
		return id;
	}
	public String getName()
	{
		return name;
	}
	public static ExportFormats fromInt(int id)
	{
		for(ExportFormats format : values())
		{
			if(format.getId() == id)
				return format;
		}
		return None;
	}
	public static ExportFormats fromName(String name)
	{
		for(ExportFormats format : values())
		{
			if(format.getName() == name)
				return format;
		}
		return None;
	}
}
